### DISCUSSION TITLE
Implement Support for Blender's Grease Pencil

### DISCUSSION THREAD

**Discussion by CarpenterBlue** [godotengine/godot-proposals#6815] [3D]
### Describe the project you are working on

A game that uses 2.5d for enemies.

### Describe the problem or limitation you are having in your project

We have to design everything in Clip Studio Paint, export each layer separately, import into Blender, position them correctly, rig, animate and import into Godot.
Lot of work, relaying on proprietary tools.
  

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Godot could use Grease Pencil for NPR rendering and animation. Right now, Blender's Grease Pencil is amazing tool that is not utilized in any other game engine even though it seems like a lighter alternative to sprite based animation. Godot could be first and protentional a trend setter as it would open up doors for the Godot users to use this amazing free and open source tool in their games, a great Feature to have that could separate it from other game engines.

Suddenly, we could design Intricate UI designs in Blender, draw worlds and locations, animate characters and straight up do frame by frame animation in a dimension agnostic way. It would be absolute boon for indie devs of any kind. 

https://user-images.githubusercontent.com/114881518/236446565-38c2e8fb-b5e5-4077-9a97-06f76a7f2e00.mp4
![image](https://user-images.githubusercontent.com/114881518/236446781-7dab7284-dd8d-45da-997f-02e585568c38.png)

(art by: [MagicianWorldly9187](https://old.reddit.com/r/blender/comments/tfny3l/blender_grease_pencil_is_awesome/)

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

In collaboration with Blender, Godot could draft specifications for the exported Grease Pencil format so that it might be used outside of Blender and specifically inside of Godot and protentional other game engines.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No.

### Is there a reason why this should be core and not an add-on in the asset library?

I don't think this can be implemented as an add-on, maybe as an GDExtension.

**Comment by Calinou** [comment]
Blender code is GPL, which means that a clean-room reimplementation must be done. This is a lot of work, especially when there doesn't seem to be a standard format for storing Grease Pencil data.

> I don't think this can be implemented as an add-on, maybe as an GDExtension.

This can be an add-on 🙂

**Comment by blurymind** [comment]
could be a nice free alternative to spine2d

**Reply by CarpenterBlue** [reply]
I gave up on doing anything 3D with Godot, the level editing tools required for that are straight up non-existent. Closing this as It is guaranteed to never happen.

**Comment by BluntBSE** [comment]
Would be really incredible, honestly.

**Reply by Calinou** [reply]
@BluntBSE Please don't bump issues without contributing significant new information. Use the :+1: reaction button on the first post instead.
